### PR TITLE
[Languages] OOP Concept: Understanding Inheritance and Polymorphism in Java

### LINKED ISSUE #3987: [Languages] OOP Concept: Understanding Inheritance and Polymorphism in Java
#### Introduction
Java is object-oriented programming, which means everything is considered an object. 
There are about four major concepts in the OOP programming paradigm that every developer intending to learn OOP is expected to know. 
In this tutorial, we will explore the concept of inheritance and Polymorphism.

#### Key takeaways
At the end of this tutorial, the reader should understand:
- What is polymorphism
- What is an inheritance
- Multiple inheritance
- The importance of polymorphism and inheritance
- Inheritance hierarchy
- Implementation of inheritance and polymorphism in Java

#### Article quality
Since I will explore these concepts in-depth using easy-to-understand English language and simple examples, this article will be a beginner-friendly article and also useful for already established java developers. This can be seen as a continuation of this article I wrote [oop concepts](https://www.section.io/engineering-education/oop-in-java-abstraction-and-encapsulation/)

#### Reference
[oop concepts](https://www.section.io/engineering-education/oop-in-java-abstraction-and-encapsulation/)

### PR DESCRIPTION
**Pull request by Frank-dev20**
This closes #3987

### CODE REVIEW

File: content/articles/understanding-java-inheritance-and-polymorphism/index.md [hunk 1]
```diff
@@ -0,0 +1,336 @@
+### Introduction
+ 
+Java is an object-oriented programming language since it provides Java developers with the ability to implement objects as real-life entities. OOP is a programming paradigm that is built around objects and their properties (attributes) and behaviors (methods). So, generally, software design using object-oriented principles is focused on objects and operations on them. Amongst many features of object-oriented programming is the fact that it encourages code reusability and extensibility. We will explain these two terms in detail later in this article.
```
Comment (john-amiscaray) [inline content/articles/understanding-java-inheritance-and-polymorphism/index.md:3]: You need to explain that this article will be focused solely on polymorphism and inheritance. This introduction only talks about OOP in general, making it sound like it's on OOP in general. A good intro should give a general idea about what the guide will talk about.

**Review by john-amiscaray — CHANGES_REQUESTED** [review]
Good article. I only have one minor thing I want you to add in the intro.

File: content/articles/understanding-java-inheritance-and-polymorphism/index.md (hunk 1 above)
Comment (Frank-dev20) [reply content/articles/understanding-java-inheritance-and-polymorphism/index.md]: Corrections made.

**Review by john-amiscaray — APPROVED** [review]
Nice job. Next time, however, please put your code snippets through an IDE with good Java IntelliSense or at least try to run it yourself. Some of the errors could have been found earlier in the writing process if you did this.

**Comment by john-amiscaray** [comment]
Finished the technical review, just waiting for the plagiarism check, and then I'll add the front matter and peer reviewer line, and then we'll be ready for a final review.

**Comment by Frank-dev20** [comment]
> Nice job. Next time, however, please put your code snippets through an IDE with good Java IntelliSense or at least try to run it yourself. Some of the errors could have been found earlier in the writing process if you did this.

Thank you John for bringing this to my rememberance

**Comment by LinusMuema** [comment]
plagiarism check done :heavy_check_mark:

**Comment by john-amiscaray** [comment]
@Frank-dev20 I seemed to have missed something here: **Please ensure your hero image is 600x400**.

**Comment by Frank-dev20** [comment]
> @Frank-dev20 I seemed to have missed something here: **Please ensure your hero image is 600x400**.

Doing that now

**Review by lalith1403 — APPROVED** [review]
Good article which is easy to read. Well done.

**Pull request merged**